test/test_ppx.ml

```diff
@@ -5,12 +5,25 @@
 *)
 open Tyxml_test
 
+module Dummy_html = struct
+  include HtmlWrapped
+  let p = HtmlWrapped.a
+end
```

Comment:
I think this is just testing a feature of the ppx that was previously untested. I remember discussing it with @pitag-ha back then.

Reply:
Could you be more specific ? Because that's a missfeature. The ppx use reflection to mirror the compile-time rules to the typing rules. If you rename elements like that, you break that underlying assumption.

If you replace the function, you are supposed to replace them by something with the same type.

Reply:
Ok so just summarize, you don't have a problem with the test per-se but you'd prefer if it redefined a sensible `p` function?

Reply:
Or to rephrase you'd prefer if it illustrated a proper use of of the `[%html.<module name> ...]` feature.

I'm not sure I understand what you think is a missfeature here.

Reply:
Well, for a start, I'm wondering what it's supposed to even test. The property I described is a requirement for the ppx to function. It's not a feature.

Reply:
Ah, if your goal is to test replacing the *module* (importantly, this module is supposed to respect the same signature), then sure, but we already do that.

Reply:
About the objective of the test: yes, it's to test replacing the module. I've tried to make that clear with the commit message "Add tyxml-ppx tests: (...) open modules", but maybe the term "replace" would have been better here than "open". Do you want me to change that?

About respecting the module signature: Are you referring to respecting the signature of `HtmlWrapped`? If so, I think I do that, because `p` is a value in `HtmlWrapped` of the same type as `a`. So my intention was to shadow the definition of `p` with another definition of the same type.

Which are the tests you're referring to that already test replacing the module? Before writing this test, my changes already passed the tests without taking the replacing module feature into account. That's why I added it: I just wanted to make sure I wouldn't break that feature when integrating it with ppxlib. I added the test to the PR, because I thought  it might also be helpful for others. But if you don't agree, I can get rid of it.

Reply:
In `test_ppx` we use both `Html` and `HtmlWrapped`, so the fact that we can substitute the module is already tested, but it might only test in one direction. I'm curious how your changes made the test pass while breaking that feature. 

`p` and `a` do not have the same types, so that substitution isn't valid. For something simple that covers the same need, I would prefer `let a ~a (elt : Html_types.a_content_fun list) = Html.a ~a (txt "a link : " :: elt)`, which is always correct, and of the same type.

Reply:
> I'm curious how your changes made the test pass while breaking that feature.

I first ported the rest ignoring that feature by always passing `None` to `get_modname` which (when implementing the feature) should only be done if there's no module in the extension point name. I was expecting the tests to fail, but they passed.



> For something simple that covers the same need, I would prefer `let a ~a (elt : Html_types.a_content_fun list) = Html.a ~a (txt "a link : " :: elt)`

That gives me the type error "Error: This expression has type string but an expression was expected of type
         string wrap". Do you know how to solve it?

Reply:
I pushed a fix for the test so that it behaves more to my linking, while still testing what you suggested.